pootle/apps/virtualfolder/models.py

```diff
@@ -131,74 +123,8 @@ def save(self, *args, **kwargs):
         self.clean_fields()
 
         self.name = self.name.lower()
-
-        if self.pk is None:
-            projects = set()
-        else:
-            # If this is an already existing vfolder, keep a list of the
-            # projects it was related to.
-            projects = set(Project.objects.filter(
-                translationproject__stores__unit__vfolders=self
-            ).distinct().values_list('code', flat=True))
-
         super(VirtualFolder, self).save(*args, **kwargs)
 
-        # Clean any existing relationship between units and this vfolder.
-        self.units.clear()
-
-        # Recreate relationships between this vfolder and units.
-        vfolder_stores_set = set()
-
-        for location in self.all_locations:
-            for filename in self.filter_rules.split(","):
-                vf_file = "".join([location, filename])
-
-                qs = Store.objects.live().filter(pootle_path=vf_file)
-
-                if qs.exists():
-                    self.units.add(*qs[0].units.all())
-                    vfolder_stores_set.add(qs[0])
-                else:
-                    if not vf_file.endswith("/"):
-                        vf_file += "/"
-
-                    if Directory.objects.filter(pootle_path=vf_file).exists():
-                        qs = Unit.objects.filter(
-                            state__gt=OBSOLETE,
-                            store__pootle_path__startswith=vf_file
-                        )
-                        self.units.add(*qs)
-                        vfolder_stores_set.update(Store.objects.filter(
-                            pootle_path__startswith=vf_file
-                        ))
-
-        # For each store create all VirtualFolderTreeItem tree structure up to
-        # its adjusted vfolder location.
-        for store in vfolder_stores_set:
-            try:
-                VirtualFolderTreeItem.objects.get_or_create(
```

Comment:
Just discussed this. This is now being handled by signals `VirtualFolder.post_save` and `Store.post_save`.